tests/testsuite/publish.rs

```diff
@@ -1686,6 +1686,410 @@ repository = "foo"
+",
+        )
+        .run();
+
+    publish::validate_upload_with_contents(
+        r#"
+        {
+          "authors": [],
+          "badges": {},
+          "categories": [],
+          "deps": [
+              {
+                "default_features": true,
+                "features": [
+                  "cat"
+                ],
+                "kind": "normal",
+                "name": "bat",
+                "optional": false,
+                "target": null,
+                "version_req": "^1.0"
+              },
+              {
+                "default_features": true,
+                "features": [
+                  "f1"
+                ],
+                "kind": "normal",
+                "name": "common",
+                "optional": false,
+                "target": null,
+                "version_req": "^1.0"
+              },
+              {
+                "default_features": true,
+                "features": [
+                  "cat"
+                ],
+                "kind": "dev",
+                "name": "bat",
+                "optional": false,
+                "target": null,
+                "version_req": "^1.0"
+              },
+              {
+                "default_features": true,
+                "features": [
+                  "cat"
+                ],
+                "kind": "build",
+                "name": "baz",
+                "optional": false,
+                "target": null,
+                "version_req": "^1.0"
+              }
+          ],
+          "description": "foo",
+          "documentation": "foo",
+          "features": {
+            "foo_feature": [
+                "common/f1",
+                "baz/cat",
+                "bat/cat"
+            ]
+          },
+          "homepage": "foo",
+          "keywords": [],
+          "license": "MIT",
+          "license_file": null,
+          "links": null,
+          "name": "foo",
+          "readme": null,
+          "readme_file": null,
+          "repository": "foo",
+          "rust_version": null,
+          "vers": "0.1.0"
+        }
+        "#,
+        "foo-0.1.0.crate",
+        &["Cargo.lock", "Cargo.toml", "Cargo.toml.orig", "src/main.rs"],
+        &[(
+            "Cargo.toml",
+            &format!(
+                r#"{}
+[package]
+edition = "2015"
+name = "foo"
+version = "0.1.0"
+authors = []
+description = "foo"
+homepage = "foo"
+documentation = "foo"
+license = "MIT"
+repository = "foo"
+
+[dependencies.bat]
+version = "1.0"
+features = ["cat"]
+
+[dependencies.common]
+version = "1.0"
+features = ["f1"]
+
+[dev-dependencies.bat]
+version = "1.0"
+features = ["cat"]
+
+[build-dependencies.baz]
+version = "1.0"
+features = ["cat"]
+
+[features]
+foo_feature = [
+    "common/f1",
+    "baz/cat",
+    "bat/cat",
+]
+"#,
+                cargo::core::package::MANIFEST_PREAMBLE
+            ),
+        )],
+    );
+}
+
+#[cargo_test]
+fn publish_with_feature_point_to_target_dev_dep() {
```

Comment:
Do we need two separate tests or could just one do?

Reply:
i'll update to one test. 
Fixed.